zebra/zebra_dplane.c

```diff
@@ -4683,6 +4687,13 @@ static int dplane_thread_loop(struct thread *event)
 		DPLANE_UNLOCK();
 	}
 
+	/*
+	 * We hit the work limit while processing at least one provider's
+	 * output queue - ensure we come back and finish it.
+	 */
+	if (reschedule)
```

Comment:
Shouldn't we only be rescheduling *if* we still have work on the zdplane_info.dg_update_ctx_q?
Seems like the rescheduling should be a `if (TAILQ_FIRST(&zdplane_info.dg_update_ctx_q)) reschedule = true;` around 4576.

Reply:
No, we need to reschedule when there is work in any provider's output queue eg. `prov->dp_ctx_out_q` rather than only when there is incoming work in `zdplane_info.dg_update_ctx_q`.

We could indeed check if there is actually anything in the output queue, however this would only really optimise the case where the queue(s) all contained exactly `limit` items and are thus now empty. This check is an [existing pattern](https://github.com/FRRouting/frr/blob/cec0e51e25d3a9049330dbeab4de2d4210937600/zebra/zebra_dplane.c#L4224) I re-used here. Happy to change it.